app/Utils/HtmlEngine.php

```diff
@@ -221,16 +221,20 @@ public function buildEntityDataArray(): array
 
             if ($this->entity->project) {
                 $data['$project.name'] = ['value' => $this->entity->project->name, 'label' => ctrans('texts.project')];
-                $data['$invoice.project'] = &$data['$project.name'];
-                $data['$quote.project'] = &$data['$project.name'];
+            } else {
+                $data['$project.name'] = ['value' => '', 'label' => ''];
```

Comment:
instead of the if / else here, just define these empty entries at the top of the file somewhere, that way they will only be overwritten if the name exists